### PR TITLE
JIR-11261 : Zoom meeting SDK upgrade to 5.15.7 for android and iOS

### PR DESCRIPTION
**Pull request by NehalDutta**
**MR change details**

- With this MR, we have updated Zoom Meeting SDK to 5.15.7 version for both Android and iOS for PCM/T applications.
- Mainly the changes were in the way how Zoom SDK will be initialized with the new meeting SDK.
- Earlier zoom SDK used appkey and secretkey to initialise. Now this is done based on the JWT token instead.
- We have updated meeting SDKs for both the platforms android & iOS
- Made changes to version.txt and plugin.xml for version change as per the current meeting SDK
- Added new method to zoom.js class for initialising with JWT token

### CODE REVIEW

File: libs/android/commonlib.aar
Comment (NehalDutta) [inline libs/android/commonlib.aar:1]: Updated Android Meeting SDK to 5.15.7

File: libs/android/mobilertc.aar
Comment (NehalDutta) [inline libs/android/mobilertc.aar:1]: Updated Android Meeting SDK to 5.15.7

File: plugin.xml
Comment (NehalDutta) [inline plugin.xml:1]: Updated version number

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "cordova.plugin.zoom-auto-audio-and-video",
-  "version": "4.6.21666.0518",
+  "version": "5.15.7",
```
Comment (NehalDutta) [inline package.json:3]: Updated version number

File: plugin.xml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <?xml version='1.0' encoding='utf-8'?>
-<plugin id="cordova.plugin.zoom-auto-audio-and-video" version="4.6.21666.0520" xmlns="http://apache.org/cordova/ns/plugins/1.0" xmlns:android="http://schemas.android.com/apk/res/android">
+<plugin id="cordova.plugin.zoom-auto-audio-and-video" version="5.15.7" xmlns="http://apache.org/cordova/ns/plugins/1.0" xmlns:android="http://schemas.android.com/apk/res/android">
```
Comment (NehalDutta) [inline plugin.xml:2]: Updated version number

File: src/android/AuthThread.java [hunk 3]
```diff
@@ -100,15 +100,15 @@ public void setLock(Object lock) {
...
+     * // Dev note ->  App key and secret are no longer used in SDK init params with meeting SDK 5.15.7
      * @param appKey    Zoom SDK App Key.
      * @param appSecret Zoom SDK App Secret.
      * @param webDomain Zoom SDK Web Domain. Default is "zoom.us".
      */
     public void setInitParameters(String appKey, String appSecret, String webDomain) {
         if (DEBUG) { Log.v(TAG, "Init parameter set"); }
-        this.initParams.appKey = appKey;
-        this.initParams.appSecret = appSecret;
+//        this.initParams.appKey = appKey; // With SDK 5.15.7, these params are no longer valid. We are not using this method anywhere in the code.
```
Comment (NehalDutta) [inline src/android/AuthThread.java:110]: We are actually not utilizing this method from the plugin. We have just commented out the broken code here and added dev note.

File: src/android/Zoom.java [hunk 4]
```diff
@@ -192,8 +251,9 @@ private void initialize(String appKey, String appSecret, CallbackContext callbac
         }
 
         ZoomSDKInitParams params = new ZoomSDKInitParams();
-        params.appKey = appKey;
-        params.appSecret = appSecret;
+        // These are not present any more in the new SDK. use initializeWithJWT instead.
```
Comment (NehalDutta) [inline src/android/Zoom.java:254]: Kept old plugin method as it is and added Dev notes for not using it.

File: src/android/Zoom.java [hunk 5]
```diff
@@ -1258,6 +1318,11 @@ public void onMeetingUserLeave(List<Long> list) {
     @Override
     public void onMeetingUserUpdated(long l) {}
 
+    @Override
+    public void onInMeetingUserAvatarPathUpdated(long l) {
```
Comment (NehalDutta) [inline src/android/Zoom.java:1322]: Added implementation for some new interfaces which were added in the new meeting SDK

File: src/android/build-extras.gradle [hunk 6]
```diff
@@ -35,13 +35,13 @@ repositories {
     }
 }
 dependencies { 
-    implementation 'androidx.multidex:multidex:2.0.0'
+    implementation "androidx.multidex:multidex:2.0.1"
```
Comment (NehalDutta) [inline src/android/build-extras.gradle:38]: Updated dependencies as per new meeting SDK android

File: src/ios/Zoom.m [hunk 7]
```diff
@@ -48,9 +50,46 @@ - (void)initialize:(CDVInvokedUrlCommand*)command
...
+//            authService.clientKey = appKey;
+//            authService.clientSecret = appSecret;
+            // Perform SDK auth.
+            [authService sdkAuth];
+        }
+    });
+}
+
+//Added new method to set jwtToken in MobileRTCAuthService
+- (void)initializeWithJWT:(CDVInvokedUrlCommand*)command{
```
Comment (anubhav-saxena-mobile) [inline src/ios/Zoom.m:63]: Added initializeWithJWT method for initializing with jwt token.

**Pull request merged**